pypesto/history/amici.py

```diff
@@ -0,0 +1,232 @@
+        (preequilibration)
+        Takes as parameter an index or indices and returns corresponding trace
+        values. If only a single value is requested, the list is flattened.
+        """
+        return self._get_hdf5_entries(PREEQ_CPU_TIME, ix)
+
+    @trace_wrap
+    def get_preeq_timeB_trace(
+        self, ix: Union[int, Sequence[int], None] = None, trim: bool = False
+    ) -> Union[Sequence[float], float]:
+        """
+        Cumulative computation time of the steady state solver of the backward
+        problem [ms] (preequilibration).
+        Takes as parameter an index or indices and returns corresponding trace
+        values. If only a single value is requested, the list is flattened.
+        """
+        return self._get_hdf5_entries(PREEQ_CPU_TIME_B, ix)
+
+    @trace_wrap
+    def get_posteq_time_trace(
+        self, ix: Union[int, Sequence[int], None] = None, trim: bool = False
+    ) -> Union[Sequence[float], float]:
+        """
+        Cumulative computation time of the steady state solver [ms]
+        (postequilibration).
+        Takes as parameter an index or indices and returns corresponding trace
+        values. If only a single value is requested, the list is flattened.
+        """
+        return self._get_hdf5_entries(POSTEQ_CPU_TIME, ix)
+
+    @trace_wrap
+    def get_posteq_timeB_trace(
+        self, ix: Union[int, Sequence[int], None] = None, trim: bool = False
+    ) -> Union[Sequence[float], float]:
+        """
+        Cumulative computation time of the steady state solver of the backward
+        problem [ms] (postequilibration).
+        Takes as parameter an index or indices and returns corresponding trace
+        values. If only a single value is requested, the list is flattened.
+        """
+        return self._get_hdf5_entries(POSTEQ_CPU_TIME_B, ix)
+
+
+class CsvAmiciHistory(CsvHistory):
+    """Stores a representation of the history in a CSV file, extended with
+    AMICI-specific traces of total simulation time, pre-equilibration time and
+    post-equilibration time.
+
+    Parameters
+    ----------
+    file:
+        CSV file name.
+    x_names:
+        Parameter names.
+    options:
+        History options.
+    load_from_file:
+        If True, history will be initialized from data in the specified file.
+    """
+
+    def __init__(
+        self,
+        file: str,
+        x_names: Sequence[str] = None,
+        options: Union[HistoryOptions, dict] = None,
+        load_from_file: bool = False,
+    ):
+        super().__init__(file, x_names, options, load_from_file=load_from_file)
+
+    def _trace_columns(self) -> list[tuple]:
+        columns = super()._trace_columns()
+        return columns + [
+            (c, np.nan)
+            for c in [
+                CPU_TIME_TOTAL,
+                PREEQ_CPU_TIME,
+                PREEQ_CPU_TIME_B,
+                POSTEQ_CPU_TIME,
+                POSTEQ_CPU_TIME_B
+            ]
+        ]
```

Comment:
Should this also exist for `Hdf5AmiciHistory`?

Reply:
This is for `_init_trace which is only needed in Csv case, I think